src/validation.cpp

```diff
@@ -2343,6 +2343,9 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
         if (pindexFork != pindexNewTip) {
             uiInterface.NotifyBlockTip(fInitialDownload, pindexNewTip);
         }
+
+        int nStopAtHeight = GetArg("-stopatheight", DEFAULT_STOPATHEIGHT);
```

Comment:
Can you move this line out of the loop? No need to parse the command line argument for every block.